### PR TITLE
Docker dev

### PR DESCRIPTION
**Pull request by Eidenn005**
Add Docker support
Added Dockerfile and docker-compose
Added dependency

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/repository/CartItemRepository.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package com.example.onlinebookstore.repository;
 
 import com.example.onlinebookstore.model.CartItem;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    @EntityGraph(value = "book.id")
```
Comment (karma-o) [inline src/main/java/com/example/onlinebookstore/repository/CartItemRepository.java]: I am not sure you want to set only ID here, it is likely that you need either the whole book or nothing from it, I think ID is fetched by default, please debug and check it in accordance with what should return your endpoint that uses this method

File: src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -39,7 +39,6 @@ public UserResponseDto save(
                 () -> new IllegalStateException("Default role not found"));
         user.setRoles(Collections.singleton(defaultRole));
         User savedUser = userRepository.save(user);
-        shoppingCartService.createForUser(savedUser);
```
Comment (karma-o) [inline src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java]: why remove it?

File: pom.xml [hunk 3]
```diff
@@ -104,6 +104,10 @@
 			<version>${jjwt.version}</version>
 			<scope>runtime</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-docker-compose</artifactId>
+		</dependency>
```
Comment (karma-o) [inline pom.xml]: I am not sure we need this dependency because we do not work with compose programmatically

**Review by karma-o — CHANGES_REQUESTED** [review]
Take a look at a few comments please

File: src/main/java/com/example/onlinebookstore/repository/CartItemRepository.java (hunk 1 above)
Comment (Eidenn005) [reply src/main/java/com/example/onlinebookstore/repository/CartItemRepository.java]: Remove.

File: src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (Eidenn005) [reply src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java]: I think accidentally while resolving conflicts

File: pom.xml (hunk 3 above)
Comment (Eidenn005) [reply pom.xml]: removed

**Review by karma-o — APPROVED** [review]
GJ!

**Pull request merged**